src/NServiceBus.Persistence.AzureStorage/Timeout/TimeoutLogic/TimeoutPersister.cs

```diff
@@ -50,19 +50,12 @@ public TimeoutPersister(string timeoutConnectionString, string timeoutDataTableN
         {
             var timeoutDataTable = client.GetTableReference(timeoutDataTableName);
 
-            timeout.Headers.TryGetValue(Headers.MessageId, out var identifier);
-            if (string.IsNullOrEmpty(identifier))
-            {
-                identifier = Guid.NewGuid().ToString();
-            }
+            var identifier = Guid.NewGuid().ToString();
             timeout.Id = identifier;
 
-            var timeoutDataEntity = await GetTimeoutData(timeoutDataTable, identifier, string.Empty).ConfigureAwait(false);
-            if (timeoutDataEntity != null) return;
-
             var headers = Serialize(timeout.Headers);
 
-            var saveActions = new List<Task>
+            var saveActions = new List<Task>(3)
```

Comment:
nitpicking, but wouldn't this be easier as an array?

Reply:
Thought about that and left it because below I use similar code which clears the list. More cumbersome with arrays. I felt consistency in the same file wouldnt hurt

Reply:
yeah, definitely no blocker imo.